xgcm/grid.py

```diff
@@ -129,8 +129,60 @@ def __init__(
         ----------
         .. [1] Comodo Conventions https://web.archive.org/web/20160417032300/http://pycomodo.forge.imag.fr/norm.html
         """
+        if not isinstance(ds, xr.Dataset):
+            raise TypeError(
+                f"ds argument must be of type xarray.Dataset, but is of type {type(ds)}"
```

Comment:
```suggestion
                f"ds argument to `xgcm.Grid` must be of type xarray.Dataset, but is of type {type(ds)}"
```

Reply:
This makes the error unique and we can test this one separately from the axis version?